source/commands/utils/chainsmoker.ts

```diff
@@ -7,14 +7,16 @@ import mapValues from "lodash.mapvalues"
 
 export type Pattern = string
 export type Path = string
-export type KeyedPatterns<T> = { [K in Extract<keyof T, string>]: Pattern[] }
-export type KeyedPaths<T> = { [K in Extract<keyof T, string>]: Path[] }
-export type MatchResult<T> = { [K in Extract<keyof T, string>]: boolean }
+export type KeyedPatterns<T> = { readonly [K in keyof T]: Pattern[] }
+export type KeyedPaths<T> = { readonly [K in keyof T]: Path[] }
+export type _MatchResult<T> = { readonly [K in keyof T]: boolean }
```

Comment:
Slight improvement on the types here: keys are readonly; and now supports any kind of key, not only string keys. `Extract<keyof T, string>` is something I carried over from an earlier attempt at typing my own `mapValues` but no longer seems needed for this to compile.

`_MatchResult` is used in the tests.